pandas/tests/dtypes/test_dtypes.py

```diff
@@ -998,3 +998,10 @@ def test_period_dtype_compare_to_string():
     dtype = PeriodDtype(freq="M")
     assert (dtype == "period[M]") is True
     assert (dtype != "period[M]") is False
+
+
+def test_column_not_loses_categorical_when_using_loc():
```

Comment:
move to pandas/tests/frame/indexing/test_loc

Reply:
tests/indexing/test_loc.py, can put it at the bottom of TestLocSetitemWithExpansion

call it `test_loc_setitem_categorical_column_retains_dtype`

Reply:
moved it to the bottom of `TestLocSetitemWithExpansion` in `test_loc_setitem_categorical_column_retains_dtype`.